### PR TITLE
Subnautica: add logic dumper for mod

### PR DESCRIPTION
**Pull request by Berserker66**
## What is this fixing or adding?
creates the logic file needed for the Subnautica mod internal tracker to track with some logic

## How was this tested?
Switching between logical and non-logical tracker, seeing it propose different locations for different inventories and verifying that.

## If this makes graphical changes, please attach screenshots.
image

### CODE REVIEW

File: worlds/subnautica/Locations.py [hunk 1]
```diff
@@ -571,9 +571,35 @@ class LocationDict(TypedDict, total=False):
...
+    payload = {location_id: location_data["position"] for location_id, location_data in location_table.items()}
     with open("locations.json", "w") as f:
         json.dump(payload, f)
+
+    def radiated(pos: Vector):
+        aurora_dist = math.sqrt((pos["x"] - 1038.0) ** 2 + (pos["y"] - -3.4) ** 2 + (pos["y"] - -163.1) ** 2)
+        return aurora_dist < 950
+
+    def far_away(pos: Vector):
+        return (pos["x"] ** 2 + pos["z"] ** 2) > (800 ** 2)
```
Comment (black-sliver) [inline worlds/subnautica/Locations.py:588]: pos["y"] does not matter?

File: worlds/subnautica/Locations.py [hunk 2]
```diff
@@ -571,9 +571,35 @@ class LocationDict(TypedDict, total=False):
             'need_laser_cutter': False,
             'position': {'x': 83.2, 'y': -276.4, 'z': -345.5}},
 }
+
 if False:  # turn to True to export for Subnautica mod
```
Comment (black-sliver) [inline worlds/subnautica/Locations.py:575]: you could possibly have that in a separate file you execute using `python -m worlds.subnautica.export` or something - maybe as a future idea.

**Review by black-sliver — APPROVED** [review]
Besides the two comments, LGTM and rolls. I have not tested an actual game nor the locations.json

File: worlds/subnautica/Locations.py (hunk 1 above)
Comment (Berserker66) [reply worlds/subnautica/Locations.py:588]: depth is checked seperately, so yes it doesn't matter here.

**Pull request merged**